### ISSUE TITLE
sw-schedule-check-in not working for Companion Pass

### ISSUE THREAD

**Issue by danyangzhao** [DavidWittman/serverless-southwest-check-in#54]
Hi there, I'm relatively new to AWS Lambda so appreciate the help here. The setup process went smoothly but now I am stuck because it seems like the sw-receive-email function is successful in parsing my forwarded email but sw-schedule-check-in. As the result, I am not getting the confirmation email.

In Cloudwatch, I'm seeing logs for sw-receive-email with "Found reservation", but getting checkin-schedule-errors: `State changed to OK at 2019/01/08. Reason: Threshold Crossed: no datapoints were received for 1 period and 1 missing datapoint was treated as [NonBreaching].`

Anywhere I should look to diagnose where it went wrong?

**Comment by DavidWittman** [comment]
@danyangzhao Can you include the full log line from the `Found reservation` response? Feel free to anonymize any identifying info. There are some email formats which I still need to implement some updates for, so that might be the case.

**Comment by danyangzhao** [comment]
Sure, here's the line. Thanks for the help!

`[INFO]	2019-01-08T23:44:28.372Z	55b5db8b-139f-11e9-8aad-bb00554e650d	Found reservation: {'first_name': 'Danyang', 'last_name': 'Zhao', 'confirmation_number': 'SO8HXX'}
`

**Comment by DavidWittman** [comment]
Hmmm... that might be the issue, but it depends on what name you have to use to check in. You should be able to put that information into [this form](https://www.southwest.com/air/change/index.html) and pull up your reservation. Does that all look correct to you?

**Comment by DavidWittman** [comment]
How do those differ from regular itineraries? Do you check in the same way?

**Comment by danyangzhao** [comment]
Yeah the checkin is the same. However, SW does not allow you to change the itinerary without first canceling the companion flight. This [link](https://www.southwest.com/air/manage-reservation/) would work but not the one you provided.

**Comment by DavidWittman** [comment]
Interesting. When is this flight happening? I'll poke around at the API and
see if I can find somewhere to get this to work. Thanks for the feedback!

**Comment by danyangzhao** [comment]
The flight is Feb 16 so there's plenty of time. Thanks for looking into it!

**Comment by DavidWittman** [comment]
If you're reading this and you have an upcoming companion pass flight, please email me (see Github profile) so I can try to figure out how to make this work.

**Comment by zevlag** [comment]
I have an upcoming flight, next week, with a Minor Reservation linked.

Your link gives:
> Unable to modify. Minor reservation linked.
> This reservation is linked to a minor reservation. In order to change one of the linked reservations, the other must be cancelled. If you have already cancelled one of the reservations and are still unable to make a change, we may be experiencing a delay. If you need assistance, please contact a Southwest Airlines Customer Representative at 1-800-I-FLY-SWA (1-800-435-9792).


But the link @anyangzhao provided works.

How can I help?
